mrp_production_grouped_by_product/models/mrp_production.py

```diff
@@ -25,15 +26,53 @@ def _post_mo_merging_adjustments(self, vals):
             self.move_finished_ids.move_dest_ids = vals['move_dest_ids']
         self.write(new_vals)
 
-    def _find_grouping_target(self, vals):
-        mo = self.env['mrp.production'].search([
+    def _get_grouping_target_domain(self, vals):
+        """Get the domain for searching manufacturing orders that can match
+        with the criteria we want to use.
+
+        :param vals: Values dictionary of the MO to be created.
+
+        :return: Odoo domain.
+        """
+        domain = [
             ('product_id', '=', vals['product_id']),
             ('bom_id', '=', vals.get('bom_id', False)),
             ('routing_id', '=', vals.get('routing_id', False)),
             ('company_id', '=', vals.get('company_id', False)),
             ('state', '=', 'confirmed'),
-        ], limit=1)
-        return mo
+        ]
+        if not vals.get('date_planned_finished'):
+            return domain
+        date = fields.Datetime.from_string(vals['date_planned_finished'])
+        pt = self.env['stock.picking.type'].browse(vals['picking_type_id'])
```

Comment:
shouldn't the picking_type_id`be included in the domain too?

Reply:
Yes, good catch! @chienandalu please add it